src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        this[i] = '';
+      }
+
+      string += `${this[i]}${separator}`;
+    }
```

Comment:
```suggestion
      if (this[i] !== undefined || this[i] !== null) {
          string += `${this[i]}${separator}`;
      }
    }
```
Якщо використати оператор ! при перевірці, в  рядку this[i] = ''; не буде потреби.

Reply:
А, і тоді || на && замінити треба, недогледів